app/graphql/hooks.ts

```diff
@@ -67,7 +73,9 @@ const makeUseQuery =
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [queryKey, options.pause]);
 
-    return [result, executeQuery] as const;
+    const { data, ...rest } = result;
+    const res = { ...rest, data: result.queryKey === queryKey ? data : null };
+    return [res, executeQuery] as const;
```

Comment:
I am wondering if we need to set the data no null and couldn't just make the query correctly re-fetch the data in the background?

Currently it looks that once we add a dataset, whole application goes into the loading state, unmounting filters, chart, etc, while it could persist the last state and indicate loading? I think it would be less blinking on a page :)

Not related to this PR, but we also have this issue currently with changing the filters – whole chart gets unmounted and a loading indicator takes its place, when it could e.g. just change chart opacity and then update it without mount / unmount thing.

Let me know what you think :)

Reply:
Thanks 💯